### PR TITLE
Fix: handle redirected title properly before saving into database.

### PR DESCRIPTION
**Pull request by cooltey**
https://phabricator.wikimedia.org/T184828

The way of preventing saving an incorrect prefix title to the database is by calling the `/page/summary` endpoint to get the correct title. 

In this case, the user tended to store "B*-tree" article, but it is actually a section inside the "B-tree" article, and the API request will be redirected to "B-tree#Variants" and that's why the offline reading list does not work properly.

### CODE REVIEW

**Comment by sharvaniharan** [comment]
Works and is fixed.. just wondering if we can collect this information when we already make a call to pageSummary endpoint in PageFragmentLoadState, by storing it into PageViewModel on PageFragment?

File: app/src/main/java/org/wikipedia/page/PageFragment.kt [hunk 1]
```diff
@@ -1131,8 +1131,17 @@ class PageFragment : Fragment(), BackPressedHandler, CommunicationBridge.Communi
...
+            // Make sure handle redirected title before saving into database
+            disposables.add(ServiceFactory.getRest(title.wikiSite).getSummary(null, title.prefixedText)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ summary ->
+                    val finalPageTitle = summary.getPageTitle(title.wikiSite)
+                    ReadingListBehaviorsUtil.addToDefaultList(requireActivity(), finalPageTitle, source) { readingListId ->
+                        moveToReadingList(readingListId, finalPageTitle, source, false) }
+                }, {
+                    L.e(it)
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/page/PageFragment.kt:1143]: If fetching the redirected title fails for any reason, can we fall back to saving the original un-redirected title?

**Review by dbrant — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/page/PageFragment.kt (hunk 1 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/page/PageFragment.kt:1146]: Yes we can! Done.

**Review by cooltey — COMMENTED** [review]
> 
> 
> Works and is fixed.. just wondering if we can collect this information when we already make a call to pageSummary endpoint in PageFragmentLoadState, by storing it into PageViewModel on PageFragment?

The idea of calling `page/summary` is to prevent saving incorrect titles when long-pressing in-article links, so explicitly calling the API is somehow required to fix the issue.

**Pull request merged**